### PR TITLE
winbox: 3.40 -> 3.41

### PR DESCRIPTION
**Pull request by herbetom**
## Description of changes

*) added support for new internal WinBox protocol fields;

CHANGELOG: https://upgrade.mikrotik.com/routeros/winbox/3.41/CHANGELOG (Content-Type: application/octet-stream)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by drupol** [comment]
Ooh looks like we've been working on the same stuff. I should have posted my PR #329626 yesterday evening :D

File: pkgs/tools/admin/winbox/default.nix [hunk 1]
```diff
@@ -65,7 +65,7 @@ symlinkJoin {
     description = "Graphical configuration utility for RouterOS-based devices";
     homepage = "https://mikrotik.com";
     downloadPage = "https://mikrotik.com/download";
-    changelog = "https://wiki.mikrotik.com/wiki/Winbox_changelog";
+    changelog = "https://upgrade.mikrotik.com/routeros/winbox/${version}/CHANGELOG";
```
Comment (drupol) [inline pkgs/tools/admin/winbox/default.nix:68]: I prefer the old and simple HTML link, this new link forces the browser to download a file.

File: pkgs/tools/admin/winbox/default.nix [hunk 2]
```diff
@@ -10,15 +10,15 @@ let
...
 
   pname = "winbox";
-  version = "3.40";
+  version = "3.41";
   name = "${pname}-${version}";
 
   executable = fetchurl (if (wine.meta.mainProgram == "wine64") then {
     url = "https://download.mikrotik.com/routeros/winbox/${version}/winbox64.exe";
-    sha256 = "1dxny1qmq4pmdn40j9zk461p3qwwjin5d18ajhczrnqrcr2v1xwi";
+    sha256 = "sha256-i8Ps8fNZUmAOyxo4DDjIjp1jwIGjIgT9CU1YgjAHC/Y=";
```
Comment (drupol) [inline pkgs/tools/admin/winbox/default.nix:18]:
```suggestion
    hash = "sha256-i8Ps8fNZUmAOyxo4DDjIjp1jwIGjIgT9CU1YgjAHC/Y=";
```

File: pkgs/tools/admin/winbox/default.nix [hunk 3]
```diff
@@ -10,15 +10,15 @@ let
...
   } else {
     url = "https://download.mikrotik.com/routeros/winbox/${version}/winbox.exe";
-    sha256 = "11vmdkwi38y7wkdkgsqpfs4l2bdaj9yg6c8wlgfzp91227gjn5li";
+    sha256 = "sha256-NypSEC5YKpqldlkSIRFtWVD4xJZcjGcfjnphSg70wmE=";
```
Comment (drupol) [inline pkgs/tools/admin/winbox/default.nix:21]:
```suggestion
    hash = "sha256-NypSEC5YKpqldlkSIRFtWVD4xJZcjGcfjnphSg70wmE=";
```

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/tools/admin/winbox/default.nix (hunk 1 above)
Comment (herbetom) [reply pkgs/tools/admin/winbox/default.nix:68]: If you find one that's good, but i haven't found anything else with up to date infos.

(The latest changelog in the wiki is for v3.18.)

The new URL is the one beeing used internally by winbox. It's displayed if you click Tools -> Check for Updates. I would therefore expect it to be at least somewhat reliable and stable.

**Comment by herbetom** [comment]
@drupol I'm fine with closing this PR, yours looks more thorough. Would be nice if you could cherry pick/ammend/whatever your pr by the changelog change. Otherwise i'll open a seperate PR for that to discuss it further if you've got concerns or want to discuss it further.

**Pull request closed without merging**